internal/command/launch/launch.go

```diff
@@ -561,7 +553,7 @@ func run(ctx context.Context) (err error) {
 	var v2AppConfig *appv2.Config
 	if deployArgs.ForceMachines {
 
-		v2AppConfig, err = appv2.FromDefinition(api.DefinitionPtr(appConfig.Definition))
+		v2AppConfig, err = appv2.FromDefinition(api.DefinitionPtr(appConfig.RawDefinition))
```

Comment:
When `v2AppConfig` is created, its `platformVersion` isn't set, and `WriteToFile` will serialize it as a v1 configuration using its `RawDefinition`. However, `v2AppConfig`'s `RawDefinition` is an empty map, so the resulting `fly.toml` is missing almost everything:

```toml
# fly.toml file generated for a-v2-app on 2023-03-06T17:27:17-05:00

app = "a-v2-app"
primary_region = "ord"
```

Reply:
Hey, I think I addressed it in today's fixes.

1. Retain RawDefinition on FromDefinition call https://github.com/superfly/flyctl/pull/1830/commits/d61015b8b634c9b654217a9d5bbce11a82e693fe

2. https://github.com/superfly/flyctl/pull/1830/commits/8da4db84f412740e988151628ab85fe4c4a1d2f5

Reply:
Awesome! Works correctly for me now, and also looks like #1820 is fixed.